tests/NovelRT.Tests/Animation/SpriteAnimatorStateTest.cpp

```diff
@@ -0,0 +1,110 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT License (MIT). See LICENCE.md in the repository root for more information.
+
+#include <gtest/gtest.h>
+#include <NovelRT.h>
+
+using namespace NovelRT;
+using namespace NovelRT::Maths;
+using namespace NovelRT::Animation;
+
+class SpriteAnimatorStateTest : public testing::Test {
+protected:
+  std::vector<std::shared_ptr<SpriteAnimatorState>> _states;
+
+  void SetUp() override {
+    _states = std::vector<std::shared_ptr<SpriteAnimatorState>>{
```

Comment:
What is the behavior if we use the constructor that takes the initial size?
It looks like it calls `std::shared_ptr<SpriteAnimatorState>()`, which would give us a `shared_ptr` with no backing allocation, in which case we could delay allocating until it is needed?

Reply:
I missed that this was a tests file, you can ignore :)

Reply:
I am unsure. I just did this way so it was explicitly clear what I wanted.